src/test/java/io/bf2/kafka/authorizer/CustomAclAuthorizerTest.java

```diff
@@ -298,28 +268,57 @@ void testPartitionLimitEnforcementFeatureFlag(String featureFlag, AuthorizationR
 
             auth.configure(customConfig);
 
-            AuthorizableRequestContext rc = Mockito.mock(AuthorizableRequestContext.class);
-            Mockito.when(rc.clientAddress()).thenReturn(InetAddress.getLoopbackAddress());
-            Mockito.when(rc.listenerName()).thenReturn("security-9095");
-            Mockito.when(rc.principal()).thenReturn(new KafkaPrincipal(KafkaPrincipal.USER_TYPE, "admin"));
-            Mockito.when(rc.requestType()).thenReturn(37);
+            AuthorizableRequestContext rc = createMockRequestContext("security-9095", KafkaPrincipal.USER_TYPE, "admin", ApiKeys.CREATE_PARTITIONS.id);
 
             Action action = new Action(AclOperation.ALTER, new ResourcePattern(ResourceType.TOPIC, "foo", PatternType.LITERAL), 0, true, true);
-            List<AuthorizationResult> results = auth.authorize(rc, Arrays.asList(action));
+            List<AuthorizationResult> results = auth.authorize(rc, List.of(action));
 
             assertEquals(1, results.size());
             assertEquals(result, results.get(0));
         }
     }
 
+
+    private void createAclsAndExpect(String user, List<AclBinding> bindings, Predicate<AclCreateResult> allMatch, int expectedResults) throws IOException {
+        try (CustomAclAuthorizer auth = new CustomAclAuthorizer(this.delegate)) {
+            auth.configure(config);
+
+            AuthorizableRequestContext rc = createMockRequestContext("security-9095", KafkaPrincipal.USER_TYPE, user, ApiKeys.CREATE_ACLS.id);
+
+            var results = auth.createAcls(rc, bindings);
+
+            List<AclCreateResult> createResults = results.stream()
+                    .map(CompletionStage::toCompletableFuture)
+                    .map(CompletableFuture::join)
+                    .collect(Collectors.toList());
+
+            assertEquals(expectedResults, createResults.size());
+            assertTrue(createResults.stream().allMatch(allMatch));
+        }
+    }
+
+    private static AuthorizableRequestContext createMockRequestContext(String listener1, String principalType, String principalName, int requestType1) {
```

Comment:
nit: could we remove the number at the end of param? ex: listener1 -> listener?

Reply:
fixed